functional-programming-spreadsheet/index200.js

```diff
@@ -0,0 +1,33 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const spreadsheetFunctions = {
+  "": x => x
+};
+
+const applyFn = str => {
+  const noHigh = highPrecedence(str);
+  const infix = /([0-9.]+)([+-])([0-9.]+)/;
+  const str2 = infixEval(noHigh, infix);
+  const regex = /([a-z]*)\(([0-9., ]*)\)(?!.*\()/i;
+}
+
+/*
```

Comment:
Eventually we'd like to have the commented out test descriptions right above or below the area where the change will take place. It's not important now, but I can see this step eventually looking something like this:

```js
...
const applyFn = str => {
  const noHigh = highPrecedence(str);
  const infix = /([0-9.]+)([+-])([0-9.]+)/;
  const str2 = infixEval(noHigh, infix);
  const regex = /([a-z]*)\(([0-9., ]*)\)(?!.*\()/i;
  /*
  The `split` method returns an array of strings from a larger string by using its argument to determine where to make each split.
  Add a function `toNumberList` which takes an argument `args` and splits it by commas.
  */

}
```

Learners will be somewhat familiar with `split` by now, though it would be good to show a simple example of splitting on another character and the output they can expect.